### PR TITLE
[DPC-3635] Add Sidekiq configuration

### PR DESCRIPTION
**Pull request by ashley-weaver**
## 🎫 Ticket

https://jira.cms.gov/browse/DPC-3635

## 🛠 Changes

Added Sidekiq to dpc-portal.

## ℹ️ Context for reviewers

(Background context, more in-depth details of the implementation, and anything else you'd like to call out or ask reviewers.)

## ✅ Acceptance Validation

(How were the changes verified? Did you fully test the acceptance criteria in the ticket? Provide reproducible testing instructions and screenshots if applicable.)

## 🔒 Security Implications

- [ ] This PR adds a new software dependency or dependencies.
- [ ] This PR modifies or invalidates one or more of our security controls.
- [ ] This PR stores or transmits data that was not stored or transmitted before.
- [ ] This PR requires additional review of its security implications for other reasons.

If any security implications apply, add Jason Ashbaugh (GitHub username: StewGoin) as a reviewer and do not merge this PR without his approval.

### CODE REVIEW

File: dpc-portal/config/initializers/sidekiq.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    def call(severity, time, program_name, message)
+      log_msg = JSON.parse(super)
+      log_msg['process'] = 'sidekiq-server'
+      Sidekiq.dump_json(log_msg) << "\n"
+    end
+  end
+
+  config.logger.formatter = DPCJSON.new
+
+  config.redis = { url: "#{ENV.fetch('REDIS_URL', 'redis://localhost')}:6379/1" }
```
Comment (chris-ronning-ny) [inline dpc-portal/config/initializers/sidekiq.rb:17]: Can we just confirm that we actually want to be hitting the same redis port for this as we are for the [admin portal](https://github.com/CMSgov/dpc-app/blob/master/dpc-admin/config/initializers/sidekiq.rb). CC @kyeah

**Review by chris-ronning-ny — APPROVED** [review]
Looks good to me. Just one comment and thing that might use clarification. If that's already something you know for sure, then this is good to go.

File: dpc-portal/config/initializers/sidekiq.rb (hunk 1 above)
Comment (ashley-weaver) [reply dpc-portal/config/initializers/sidekiq.rb:17]: fwiw, it's the same port we're hitting in [dpc-web](https://github.com/CMSgov/dpc-app/blob/master/dpc-web/config/initializers/sidekiq.rb) too

**Review by jdettmannnava — APPROVED** [review]
Looks good to me!

File: dpc-portal/config/initializers/sidekiq.rb (hunk 1 above)
Comment (kyeah) [reply dpc-portal/config/initializers/sidekiq.rb:17]: Good question; yes, we should attach to the same port (6379 is the default port that Redis listens in on.) 

If we want to use a separate Redis instance entirely in deployed environments, we can change that later in the terraform by changing the REDIS_URL.

**Pull request merged**